tests/jobs/test_scheduler_job.py

```diff
@@ -188,6 +188,55 @@ def test_process_executor_events(self, mock_stats_incr):
 
         mock_stats_incr.assert_called_once_with('scheduler.tasks.killed_externally')
 
+    def test_scheduler_executor_overflow(self):
+        """
+        Test that tasks that tasks that are set back to scheduled after
+        """
+        executor = TestExecutor(do_update=True, parallelism=3)
+        session = settings.Session()
+        dag = self.create_test_dag()
+        task = DummyOperator(
+            task_id='dummy',
+            dag=dag,
+            owner='airflow')
+        from datetime import timedelta
+        tis = []
+        for i in range(1,10):
+            ti = TI(task, DEFAULT_DATE + timedelta(days=i))
+            ti.state = State.SCHEDULED
+            tis.append(ti)
+            session.merge(ti)
+        session.commit()
+
+        simple_dag_bag = self._make_simple_dag_bag([dag])
+        scheduler = SchedulerJob(num_runs=1,
+                                 executor=executor,
+                                 subdir=os.path.join(settings.DAGS_FOLDER,
+                                                     "no_dags.py"))
+        mock.patch.object(scheduler, '_change_state_for_tis_without_dagrun').start()
```

Comment:
Do we need to patch this?

Reply:
Yes. Otherwise the tasks won't run because they don't have an associated DagRun.